### DISCUSSION TITLE
Docker option to make self signed cert

### DISCUSSION THREAD

**Discussion by bmartino1** [rejetto/hfs#801] [General]
Thought I'd Ask:
https://github.com/rejetto/hfs/discussions?discussions_q=cert
as I'm not seeing any other discussion on this. Not that important as end user should make this by them selves...

as seen here in the repo readme for Unraid.
https://github.com/bmartino1/hfs-docker/blob/main/README.md

once I have end users enter the admin interface:

![image](https://github.com/user-attachments/assets/3ec0f383-a972-433f-9762-c965ef08b476)

Then click on make a self singed certificate. once I get the Certificate saved the docker appears to break. netwrok error for anything hfs from home page to admin interface.

![image](https://github.com/user-attachments/assets/21c8ce9a-5776-4e66-8246-c0d2cb277cc2)

As stop and start of the docker fixes this and then https uses the self-signed certificate.

Is there a docker value that I can run that will generate this at first run?

**Comment by rejetto** [comment]
nope,
but it would be nice to know more of the problem.
did you look at the console?

**Comment by bmartino1** [comment]
I have not this is what i have to reproduce... I get connection error... I
think this is because of the https force option being set before s https
cert is made

Pulling image: rejetto/hfs:latest

IMAGE ID [1822495604]: Pulling from rejetto/hfs.
IMAGE ID [43c4264eed91]: Already exists.
IMAGE ID [9ff590c4b6e1]: Pulling fs layer. Downloading 100% of 93 B. Verifying
Checksum. Download complete. Extracting. Pull complete.
IMAGE ID [9c580c71bd25]: Pulling fs layer. Downloading 100% of 3 MB. Verifying
Checksum. Download complete. Extracting. Pull complete.
IMAGE ID [7a7aa684fc25]: Pulling fs layer. Downloading 100% of 989 B. Verifying
Checksum. Download complete. Extracting. Pull complete.
IMAGE ID [30f3a14b812e]: Pulling fs layer. Downloading 100% of 25 MB. Verifying
Checksum. Download complete. Extracting. Pull complete.
Status: Downloaded newer image for rejetto/hfs:latest

*TOTAL DATA PULLED:* 28 MB

Command executiondocker run
  -d
  --name='RejettoHFS3'
  --net='bond0'
  --ip='192.168.2.50'
  --pids-limit 2048
  -e TZ="America/Chicago"
  -e HOST_OS="Unraid"
  -e HOST_HOSTNAME="BMM-Unraid"
  -e HOST_CONTAINERNAME="RejettoHFS3"
  -e 'HFS_CREATE_ADMIN'=''
  -e 'HFS_LOG_ROTATION'='weekly'
  -e 'HFS_FORCE_HTTPS'='true'
  -e 'HFS_ADMIN_NET'='192.168.2.0/24'
  -e 'HFS_VFS'='{"source": "/app/myDisk"}'
  -e 'HFS_HTTPS_PORT'='443'
  -e 'HFS_PORT'='80'
  -e 'HFS_CERT'='/home/hfs/.hfs/self.cer'
  -e 'HFS_ERROR_LOG'='/app/logs/error.log'
  -e 'HFS_PRIVATE_KEY'='/home/hfs/.hfs/self.key'
  -e 'HFS_LOG'='/app/logs/access.log'
  -l net.unraid.docker.managed=dockerman
  -l net.unraid.docker.webui='http://[IP]:[PORT:80]/~/admin/'
  -l net.unraid.docker.icon='
https://raw.githubusercontent.com/bmartino1/unraid-docker-templates/refs/heads/main/images/rejettohfs.png
'
  -v '/mnt/user/appdata/hfs-docker/hfsconf/':'/home/hfs/.hfs':'rw'
  -v '/mnt/user/appdata/hfs-docker/certs/':'/app/certs':'rw'
  -v '/mnt/user/appdata/hfs-docker/logs/':'/app/logs':'rw'
  -v '/mnt/user/Users/':'/app/myDisk':'rw'
  --hostname HFS3
  --mac-address 02:42:c0:a8:02:D4 'rejetto/hfs:latest'

[image: image.png]


[image: image.png]
[image: image.png]

[image: image.png]

**Reply by bmartino1** [reply]
Just to note. 
if I stop the docker then start the docker, everything works as it should.

Just thought id share here. image sent vie email.

**Comment by bmartino1** [comment]
so seeing as i'm getting a conection err lest stop the docekr and see whats in consloe:

![image](https://github.com/user-attachments/assets/48dbe830-a32e-4492-a68b-ee20d62f5cdf)

![image](https://github.com/user-attachments/assets/e16918ec-3e8b-4740-9eff-4af7b8556169)

![image](https://github.com/user-attachments/assets/f6606765-b6ab-4251-82b2-b48a811127e3)


docekr log as it looks like consol is the same as unraid webui log:

16:22:29 cannot launch browser on this machine >PLEASE< open your browser and reach one of these (you may need a different address) 
 - http://192.168.2.50/~/admin/ 
 - http://104.219.38.175/~/admin/
16:38:23 cannot launch browser on this machine >PLEASE< open your browser and reach one of these (you may need a different address) 
 - http://192.168.2.50/~/admin/ 
 - http://104.219.38.175/~/admin/
16:22:29 HFS ~ HTTP File Server
16:22:29 © Massimo Melina <a@rejetto.com> - License https://www.gnu.org/licenses/gpl-3.0.txt
16:22:29 started 11/6/2024, 4:22:29 PM 
16:22:29 version 0.53.0
16:22:29 build 2024-09-25T22:29:20.090Z
16:22:29 cwd /home/hfs/.hfs
16:22:29 app /app
16:22:29 node v18.5.0
16:22:29 platform linux x64
16:22:29 pid 1
16:22:29 config config.yaml
16:22:29 No config file, using defaults
16:22:29 HINT: type "help" for help
16:22:29 account admin created
16:22:29 http serving on any network : 80
16:22:29 serving on http://192.168.2.50
16:22:29 serving on http://104.219.38.175
16:24:35 https serving on any network : 443
16:24:35 serving on https://192.168.2.50
16:24:35 serving on https://104.219.38.175
16:38:22 HFS ~ HTTP File Server
16:38:22 © Massimo Melina <a@rejetto.com> - License https://www.gnu.org/licenses/gpl-3.0.txt
16:38:22 started 11/6/2024, 4:38:22 PM 
16:38:22 version 0.53.0
16:38:22 build 2024-09-25T22:29:20.090Z
16:38:22 cwd /home/hfs/.hfs
16:38:22 app /app
16:38:22 node v18.5.0
16:38:22 platform linux x64
16:38:22 pid 1
16:38:22 config config.yaml
16:38:23 HINT: type "help" for help
16:38:23 account admin created
16:38:23 http serving on any network : 80
16:38:23 serving on http://192.168.2.50
16:38:23 serving on http://104.219.38.175
16:38:23 https serving on any network : 443
16:38:23 serving on https://192.168.2.50
16:38:23 serving on https://104.219.38.175

**Comment by rejetto** [comment]
you may be right about the "force https" option.
try removing it.
or try manually changing the address in the browser when you get the error.

**Reply by bmartino1** [reply]
working on it will get back toyou:
docker run
  -d
  --name='RejettoHFS3'
  --net='bond0'
  --ip='192.168.2.50'
  --pids-limit 2048
  -e TZ="America/Chicago"
  -e HOST_OS="Unraid"
  -e HOST_HOSTNAME="BMM-Unraid"
  -e HOST_CONTAINERNAME="RejettoHFS3"
  -e 'HFS_CREATE_ADMIN'=''
  -e 'HFS_LOG_ROTATION'='weekly'
  -e 'HFS_ADMIN_NET'='192.168.2.0/24'
  -e 'HFS_HTTPS_PORT'='443'
  -e 'HFS_PORT'='80'
  -e 'HFS_CERT'='/home/hfs/.hfs/self.cer'
  -e 'HFS_ERROR_LOG'='/app/logs/error.log'
  -e 'HFS_PRIVATE_KEY'='/home/hfs/.hfs/self.key'
  -e 'HFS_LOG'='/app/logs/access.log'
  -l net.unraid.docker.managed=dockerman
  -l net.unraid.docker.webui='http://[IP]:[PORT:80]/~/admin/'
  -l net.unraid.docker.icon='https://raw.githubusercontent.com/bmartino1/unraid-docker-templates/refs/heads/main/images/rejettohfs.png'
  -v '/mnt/user/appdata/rejettohfs3/config':'/home/hfs/.hfs':'rw'
  -v '/mnt/user/appdata/rejettohfs3/certificate':'/app/certs':'rw'
  -v '/mnt/user/appdata/rejettohfs3/logs':'/app/logs':'rw'
  -v '/mnt/user/':'/app/myDisk':'rw'
  --hostname HFS3
  --mac-address 02:42:c0:a8:02:D4 'rejetto/hfs:latest'
  
  option removed for testing.

**Comment by bmartino1** [comment]
so even with removing the https force option 

![image](https://github.com/user-attachments/assets/8cbb6fdc-4bc1-4928-81f2-291bdb61544d)

same error after using admin interface to make slef signed cert. not sure why it fiales but if i stop and start the docekr it will be fine...

seeing as the console give me connection error:

this is the unraid docker log whcih apears to be the same data for the console log:

17:23:17 HFS ~ HTTP File Server
17:23:17 © Massimo Melina <a@rejetto.com> - License https://www.gnu.org/licenses/gpl-3.0.txt
17:23:17 started 11/6/2024, 5:23:17 PM 
17:23:17 version 0.53.0
17:23:17 build 2024-09-25T22:29:20.090Z
17:23:17 cwd /home/hfs/.hfs
17:23:17 app /app
17:23:17 node v18.5.0
17:23:17 platform linux x64
17:23:17 pid 1
17:23:17 config config.yaml
17:23:17 No config file, using defaults
17:23:17 HINT: type "help" for help
17:23:18 account admin created
17:23:18 http serving on any network : 80
17:23:18 serving on http://192.168.2.50
17:23:18 serving on http://104.219.38.175
17:23:31 https serving on any network : 443
17:23:31 serving on https://192.168.2.50
17:23:31 serving on https://104.219.38.175

![image](https://github.com/user-attachments/assets/c20e4c78-4fb6-4b50-88d1-2f67ee198cc8)

**Reply by bmartino1** [reply]
well since it wasn;t https force. It be weird as that is where the self cert is made...

but it could be that i'm calling the cert files before creation:

-e 'HFS_CERT'='/home/hfs/.hfs/self.cer'
-e 'HFS_PRIVATE_KEY'='/home/hfs/.hfs/self.key'

**Reply by bmartino1** [reply]
testing without thoeses options:

docker run
  -d
  --name='RejettoHFS3'
  --net='bond0'
  --ip='192.168.2.50'
  --pids-limit 2048
  -e TZ="America/Chicago"
  -e HOST_OS="Unraid"
  -e HOST_HOSTNAME="BMM-Unraid"
  -e HOST_CONTAINERNAME="RejettoHFS3"
  -e 'HFS_CREATE_ADMIN'=''
  -e 'HFS_LOG_ROTATION'='weekly'
  -e 'HFS_ADMIN_NET'='192.168.2.0/24'
  -e 'HFS_HTTPS_PORT'='443'
  -e 'HFS_PORT'='80'
  -e 'HFS_ERROR_LOG'='/app/logs/error.log'
  -e 'HFS_LOG'='/app/logs/access.log'
  -l net.unraid.docker.managed=dockerman
  -l net.unraid.docker.webui='http://[IP]:[PORT:80]/~/admin/'
  -l net.unraid.docker.icon='https://raw.githubusercontent.com/bmartino1/unraid-docker-templates/refs/heads/main/images/rejettohfs.png'
  -v '/mnt/user/appdata/rejettohfs3/config':'/home/hfs/.hfs':'rw'
  -v '/mnt/user/appdata/rejettohfs3/logs':'/app/logs':'rw'
  -v '/mnt/user/':'/app/myDisk':'rw'
  --hostname HFS3
  --mac-address 02:42:c0:a8:02:D4 'rejetto/hfs:latest'

**Reply by bmartino1** [reply]
welp nope.. 

![image](https://github.com/user-attachments/assets/ae843788-1a3d-42a7-8690-ca6a059ab7a3)

so i'm not sure why usign the admin interface breaks in docer...

**Reply by bmartino1** [reply]
to be extra clear i'm deleting the volumes befre to initiate a pull and docker run. it is not exiting config file issues either... so while the command works to make the self-signed cert you must stop and strat the docker to then use the https cert and hfs after a certificate is made.

**Comment by rejetto** [comment]
why the empty HFS_CREATE_ADMIN ?

**Reply by bmartino1** [reply]
security. posted here... trying to share my test passowrd account...

**Reply by rejetto** [reply]
ok, so it's not empty in the final script

**Comment by rejetto** [comment]
later i can consider a way to make the certificate at start,
but what if you do it with other tools like openssl?
and i wonder, is self-signed any good for your users?

**Reply by bmartino1** [reply]
I can use the openssl options, that is already noted and added to the template... I have a 3rd party way to use that yes. Was looking for a side fix. or docker variable to streamline the process.

Users can run as noted int he readme: 
Please Read: https://raw.githubusercontent.com/bmartino1/unraid-docker-templates/refs/heads/main/Guides/Quick%20Start%20Guide%20For%20Rejeto%20HFS.txt

openssl req -x509 -newkey rsa:4096 -keyout /mnt/user/appdata/hfs/certificate/privkey.key -out 

edit the template remove the hfs self path and use the custom one. this works...
![image](https://github.com/user-attachments/assets/fa101924-0e53-45dc-ad1c-4c8d9f289e22)
![image](https://github.com/user-attachments/assets/1cde25be-ef9b-4e83-b316-59c98d34ae79)

was more a weird bug find regarding admin interface and errors after.

**Comment by rejetto** [comment]
FYI, your values for log_rotation and "port" are the default one so, at the current time, it's the same if you remove them

**Reply by bmartino1** [reply]
was added for unraid user option to make edits and update config within the docker template.

**Comment by rejetto** [comment]
in version 0.55.0-alpha2, HFS will expose an API to create self-signed cert.

if you will call `curl localhost/~/api/make_self_signed_cert --data {\"fileName\":\"self\"}`
it will both generate the cert and will store it in the config.
